src/ECER.Resources.Documents/Applications/IApplicationRepository.cs

```diff
@@ -1,14 +1,22 @@
-﻿namespace ECER.Resources.Documents.Applications;
+﻿using ECER.Resources.Documents.PortalInvitations;
+
+namespace ECER.Resources.Documents.Applications;
 
 public interface IApplicationRepository
 {
   Task<IEnumerable<Application>> Query(ApplicationQuery query, CancellationToken cancellationToken);
 
+  Task<IEnumerable<Province>> QueryProvinces(ProvinceQuery query, CancellationToken cancellationToken);
+
   Task<string> SaveDraft(Application application, CancellationToken cancellationToken);
 
   Task<string> Submit(string applicationId, CancellationToken cancellationToken);
 
   Task<string> Cancel(string applicationId, CancellationToken cancellationToken);
+
+  Task<bool> SubmitCharacterReference(CharacterReferenceSubmissionRequest request, CancellationToken ct);
```

Comment:
no bool return value and can be consolidated into a single method with abstract ApplicationReference record

Reply:
There is a problem here, the datatype for character reference and work experience reference are completely different. we have different forms with different data models for each. we probably have to separate them.